File: march-frontend/src/Components/Bundle.js
import axios from 'axios';
import React, { useState } from 'react';

const Bundle = ({originTitle, contents, id, onDelete}) => {
    const [title, setTitle] = useState(originTitle);

    const onChangeTitle = (e) => {
        setTitle(e.target.value);
    }

    const onClickChange = () => {
        axios.put(`http://localhost:8080/march/bundle/${id}/${title}`).then(response => {
            setTitle(response.data.title);
        }).catch(e => console.log(e));
    }

    return (
        <div className="Bundle">
            <input className="bundle-title" type="text" value={title} id={id} onChange={onChangeTitle} />
            {contents}
            <button className="title-change-button" onClick={() => onClickChange(id, title)}>제목 변경</button>
            <button className="delete-button" onClick={() => onDelete(id)}>삭제 하기</button>
        </div>
    )
}


export default Bundle;

File: msa/client-service/src/main/java/com/march/clientservice/Bundle.java
package com.march.clientservice;

import lombok.Data;

import java.util.List;

@Data
public class Bundle {
    private Long id;
    private String title;
    private String contents;

    @Override
    public String toString() {
        return "{id: " + id + ", " + "title: " + title + ", " + "contents: " + contents + "}";
    }
}
